crates/fuel-core/src/service/adapters/block_importer.rs

```diff
@@ -75,6 +79,37 @@ impl BlockVerifier for VerifierAdapter {
     }
 }
 
+#[async_trait::async_trait]
+impl RelayerPort for MaybeRelayerAdapter {
+    async fn await_until_if_in_range(
+        &self,
+        da_height: &DaBlockHeight,
+        max_da_lag: &DaBlockHeight,
+    ) -> anyhow::Result<()> {
+        #[cfg(feature = "relayer")]
+        {
+            if let Some(sync) = self.relayer_synced.as_ref() {
```

Comment:
What if the relayer is still syncing for a new node? Maybe we should wait for the relayer to finish syncing with eth before checking da_height against max_da_lag?

Reply:
I guess if this fails, the stream will just start over. However, if we don't handle this case here it could be harder to determine whether the advertised da_height was malicious from the peer reputation perspective.

Reply:
The relayer is always kind of syncing so it's kind of hard to say if it's doing an initial sync. Even if we did know that I don't know what we should do differently?